client/ui/components/addon-groups/SyncDiff.jsx

```diff
@@ -0,0 +1,110 @@
+import React, { PropTypes as pt } from 'react';
+
+import RaisedButton from 'material-ui/RaisedButton';
+import { Toolbar, ToolbarGroup } from 'material-ui/Toolbar';
+
+import LoadingIndicator from '../indicators/LoadingIndicator';
+import containSyncDiff from '../../containers/SyncDiffContainer';
+import QueryAddonGroups from '../data/QueryAddonGroups';
+import goTo from '../../utils/goTo';
+
+
+const style = {
+  toolbar: {
+    justifyContent: 'flex-end',
+  },
+};
+
+
+class SyncDiff extends React.Component {
+  static propTypes = {
+    addonGroups: pt.array,
+    count: pt.number,
+    type: pt.string,
+  };
+
+  mapAddons(addonGroup) {
+    const { type } = this.props;
+
+    return addon => {
+      const diff = (type === 'qa') ? addonGroup.qa_sync_diff : addonGroup.shipped_sync_diff;
+      let classes = '';
+
+      if (diff.added.includes(addon.id)) {
+        classes = 'added';
+      } else if (diff.removed.includes(addon.id)) {
+        classes = 'removed';
+      } else if (diff.upgraded.includes(addon.id)) {
+        classes = 'upgraded';
+      } else if (diff.downgraded.includes(addon.id)) {
+        classes = 'downgraded';
+      }
+
+      return (
+        <div className={`addon ${classes}`} key={addon.id}>
+          {addon.name} <span>v{addon.version}</span>
+        </div>
+      );
+    };
+  }
+
+  renderGroups() {
+    const { addonGroups, type } = this.props;
+    const groups = [];
+
+    addonGroups.forEach(addonGroup => {
+      const synced = (type === 'qa') ? addonGroup.qa_synced : addonGroup.shipped_synced;
```

Comment:
This looks like a good use case for subclassing, so that instead of passing in a type prop, you just have two child classes that implement `synced` and `currAddons` methods based on which type of diff they're showing.